src/pages/Authentication/Signup.jsx

```diff
@@ -0,0 +1,121 @@
+import "./authentication.css";
+import { Link, useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { fetchLoginDetails } from "../../services/fetchAuth";
+import { useAuth } from "../../context";
+import { LOADING, SIGNUP, ERROR } from "../../helper/constants";
+import { Loader } from "../../components";
+
+const Signup = () => {
+  const navigate = useNavigate();
+  const { authState, authDispatch } = useAuth();
+  const { loading, error } = authState;
+  const [userName, setUserName] = useState("");
```

Comment:
Suggestion: Can create a reducer to manage all form states in both pages

Reply:
Thankyou Prabhav for this suggestion. Will keep this in mind and change this after completing my P0 features.